common/src/gas_provider/mod.rs

```diff
@@ -277,14 +273,13 @@ pub trait Provider {
     /// operations that create inequality between the underlying value
     /// supply and some hypothetical "collateral" asset.
 
-    /// `PositiveImbalance` indicates that some value has been created,
-    /// which will eventually lead to an increase in total supply.
-    type PositiveImbalance: Imbalance<Self::Balance, Opposite = Self::NegativeImbalance>;
+    /// `PositiveImbalance` indicates that some value has been added
+    /// to circulation , i.e. total supply has increased.
+    type PositiveImbalance: Imbalance<Balance = Self::Balance>;
```

Comment:
Seems that duplicate of the types here is useless, I'd go with defining only in Tree impl with Imbalance trait bound, because using Provider trait's GasTree type will always return some associated type we're not allowed to directly manipulate with, but with trait bound it's okay to use as some `impl Imbalance`

Reply:
Remove declaration from here